src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -3278,6 +3447,30 @@ TEST_F(EnsembleConfigurationValidationWithDemultiplexer, DemultiplexerNodeNotEno
     ASSERT_EQ(pipelineDefinition->validate(manager), StatusCode::PIPELINE_NOT_ENOUGH_SHAPE_DIMENSIONS_TO_DEMULTIPLY);
 }
 
+TEST_F(EnsembleConfigurationValidationWithDemultiplexer, DemultiplexerNodeNotEnoughDimensionsToDemultiply_Scalar) {
```

Comment:
What is the behavior when someone has DAG with scala input/output, and then enables demultiplexer? What are the metadata and does infer work?